app/src/androidTest/java/co/nimblehq/compose/crypto/ui/screen/HomeScreenUITest.kt

```diff
@@ -0,0 +1,185 @@
+import co.nimblehq.compose.crypto.ui.BaseScreenTest
+import co.nimblehq.compose.crypto.ui.navigation.AppDestination
+import co.nimblehq.compose.crypto.ui.screens.MainActivity
+import co.nimblehq.compose.crypto.ui.screens.home.*
+import io.kotest.matchers.shouldBe
+import io.mockk.every
+import io.mockk.mockk
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOf
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
+class HomeScreenUITest : BaseScreenTest() {
+
+    @get:Rule
+    val composeAndroidTestRule = createAndroidComposeRule<MainActivity>()
+
+    private val homeTitle: String
+        get() = composeAndroidTestRule.activity.getString(R.string.home_title)
+
+    private val totalCoinsLabel: String
+        get() = composeAndroidTestRule.activity.getString(R.string.portfolio_card_total_coin_label)
+
+    private val todayProfitLabel: String
+        get() = composeAndroidTestRule.activity.getString(R.string.portfolio_card_today_profit_label)
+
+    private val errorGeneric: String
+        get() = composeAndroidTestRule.activity.getString(R.string.error_generic)
+
+    private val expectedPriceChange: String
+        get() = composeAndroidTestRule.activity.getString(
+            R.string.coin_profit_percent,
+            MockUtil.trendingCoins.first().priceChangePercentage24hInCurrency.toFormattedString()
+        )
+
+    private val mockGetMyCoinsUseCase = mockk<GetMyCoinsUseCase>()
+    private val mockGetTrendingCoinsUseCase = mockk<GetTrendingCoinsUseCase>()
+
+    private lateinit var viewModel: HomeViewModel
+
+    private var appDestination: AppDestination? = null
+
+    @Before
+    fun setUp() {
+        composeAndroidTestRule.activity.setContent {
+            HomeScreen(
+                viewModel = viewModel,
+                navigator = { destination -> appDestination = destination }
+            )
+        }
+
+        every { mockGetMyCoinsUseCase.execute(any()) } returns flowOf(MockUtil.myCoins)
+        every { mockGetTrendingCoinsUseCase.execute(any()) } returns flowOf(MockUtil.trendingCoins)
+    }
+
+    @Test
+    fun when_enter_to_HomeScreen_it_render_the_PortfolioCard_properly() {
+        initViewModel()
+
+        with(composeAndroidTestRule) {
+            onNodeWithTag(testTag = TestTagHomeTitle).assertTextEquals(homeTitle)
+            onNodeWithTag(testTag = TestTagTotalCoinsLabel).assertTextEquals(totalCoinsLabel)
+            onNodeWithTag(testTag = TestTagTodayCoinProfitLabel).assertTextEquals(todayProfitLabel)
+            onNodeWithTag(testTag = TestTagCardTotalCoins).assertTextEquals("$7,273,291")
+            onNodeWithTag(testTag = TestTagCardTodayProfit).assertTextEquals("$193,280")
+        }
+    }
+
+    @Test
+    fun when_enter_to_HomeScreen_and_load_MyCoins_successfully_it_render_the_UI_properly() {
+        initViewModel()
+
+        with(composeAndroidTestRule) {
```

Comment:
FYI: If we compare with `HomeScreenTest`, we can't  assert this:
```kotlin
onNodeWithTag(testTag = TestTagCoinsLoader).assertIsDisplayed()
```
I'm not sure what's wrong, but if you noticed, when we run a test case, the `CircleProgress` doesn't show anymore that's why it can't find the node when we try to assert. Please let me know if you have an idea 🙏

Reply:
@Wadeewee It is because you are returning the api result immediately inside `@Before` function. So it won't show loading. Override the `composeAndroidTestRule` without passing the `viewModel` inside your loading function.
```
    @Test
    fun when_enter_to_HomeScreen_it_show_loading() {
        composeAndroidTestRule.activity.setContent {
            HomeScreen(
                navigator = { destination -> appDestination = destination }
            )
        }

        initViewModel()

        with(composeAndroidTestRule) {
            onNodeWithTag(testTag = TestTagCoinsLoader). assertIsDisplayed(homeTitle)
        }
    }
```
NOTE: We need to test the loading in a separate function because it won't work for the other tests since we didn't pass the `viewModel`.

Reply:
> It is because you are returning the api result immediately inside @Before function. So it won't show loading.

Yeah, @kaungkhantsoe, I thought it was because of this too, but It seems like it was an error on some test cases if we don't return api in the `setup` function. So strange if we compare it with the `HomeScreenTest` 🤔

Reply:
@Wadeewee I am not sure how to test with current implementation pattern of `ViewModel` and `Repository`. Normally we return generic `Resource` type which has `Loading`, `Success` and `Error` states right? That way, we can mock the return type of each api call and test the `UI`. Right now, there is no way to mock the loading state. cc @luongvo @hoangnguyen92dn

Reply:
@Wadeewee @kaungkhantsoe Regarding the loading indicator, can we have the delay on `mockGetMyCoinsUseCase.execute` to show the loading? 

```suggestion
        every { mockGetMyCoinsUseCase.execute(any()) } returns flow {
            delay(500L)
            emit(MockUtil.myCoins)
        }
```

Reply:
@hoangnguyen92dn I've tested it and it works. I didn't see the loading though 💭. But it really works 🎉

Reply:
@hoangnguyen92dn I tried to use this on [this test](https://github.com/nimblehq/android-templates/pull/399#issuecomment-1420277502) but I couldn't make it work. Can you please take a look at it? 🙏

Reply:
> I've tested it and it works. I didn't see the loading though 💭. But it really works 🎉

Nice 😄 The delay is quite fast but it's a short delay enough to keep the loading indicator shown during the test verification.

> I tried to use this on https://github.com/nimblehq/android-templates/pull/399#issuecomment-1420277502 but I couldn't make it work. Can you please take a look at it?

🤔 I don't get your idea of which test you are mentioning.

Reply:
@Wadeewee @hoangnguyen92dn yes, with Robolectric, sometimes we need to apply using `delay` or [LooperMode.PAUSED](https://robolectric.org/javadoc/4.3/org/robolectric/annotation/LooperMode.Mode.html#PAUSED). We need to recall our knowledge and experience with it. One of our achievements with Robolectric is in the ToyotaWallet project. 

https://github.com/nimblehq/toyota-wallet-android/blob/8070ca1049c672e1fdccf42849eae2f08d8a0e36/app/src/test/java/co/omise/gcpf/app/ui/screens/main/profile/methods/details/PaymentMethodDetailsFragmentTest.kt#L162-L176

Reply:
@hoangnguyen92dn Thanks for your idea 🙏 I think we need to separate the test case for each loading. Please check on this [commit](https://github.com/nimblehq/jetpack-compose-crypto/pull/86/commits/ccdf8649722015c08d0e24b91dadbf69c4e0fbb5) cc @kaungkhantsoe @luongvo